include/gatekeeper_gk.h

```diff
@@ -353,6 +353,8 @@ struct gk_flush_request {
 struct gk_cmd_entry {
 	enum gk_cmd_op  op;
 
+	uint32_t flow_hash_val;
+
 	union {
 		/* GGU policy to be added with GK_ADD_POLICY_DECISION op. */
 		struct ggu_policy ggu;
```

Comment:
1. Have `struct gk_add_policy` defined as follows to replace fields `flow_hash_val` and `ggu`.
```
struct gk_add_policy {
	struct ggu_policy policy;
	uint32_t flow_hash_val;
};
```
`struct gk_add_policy` is analogous to `struct gk_flush_request` in role.

2. The following patch, namely **gk: pass computed flow hash value to the GK block for flow logging**, needs to define the following struct:
```
struct gk_log_flow {
	struct ip_flow flow;
	uint32_t flow_hash_val;
};
```

Reply:
Should we keep ``uint32_t flow_hash_val`` in the definition of ``struct ip_flow``? In that case, even when deleting a flow entry, we can still use that hash value.

Reply:
Should we also keep ``uint32_t flow_hash_val`` in the definition of ``struct flow_entry``, in that case, even when deleting an entry from the flow hash table, we can use the hash value directly.

Reply:
Yes. It's a good idea to add `uint32_t flow_hash_val` to `struct flow_entry`.

Reply:
Thanks @AltraMayor ! I will issue a new pull request on this.

Reply:
After adding the mentioned field, there will be a compilation error on ``RTE_BUILD_BUG_ON(sizeof(*fe) > 128)``. So, the ``struct flow_entry`` is pretty large in terms of size.

Reply:
A solution to add `uint32_t flow_hash_val` to `struct flow_entry` and still keep the size of `struct flow_entry` is as follows:
1. Add `uint32_t flow_hash_val` after field `flow`.
2. Move field `u.bpf.program_index` after `in_use`. This will require a number of changes and a good comment explaining what this field is doing here, but given the benefit that we'll have with this change, it's worth it.

Reply:
In the future pull request, have a patch to move `u.bpf.program_index` and another patch to add `uint32_t flow_hash_val` to simplify the review.

Reply:
Sure, will do.